src/mass/inject.py

```diff
@@ -0,0 +1,105 @@
+# Copyright 2021 - 2023 Universität Tübingen, DKFZ, EMBL, and Universität zu Köln
+# for the German Human Genome-Phenome Archive (GHGA)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""Module hosting the dependency injection container."""
+from collections.abc import AsyncGenerator
+from contextlib import asynccontextmanager
+from typing import Optional
+
+from fastapi import FastAPI
+from ghga_service_commons.utils.context import asyncnullcontext
+from hexkit.providers.akafka.provider import KafkaEventSubscriber
+from hexkit.providers.mongodb.provider import MongoDbDaoFactory
+
+from mass.adapters.inbound.event_sub import EventSubTranslator
+from mass.adapters.inbound.fastapi_ import dummies
+from mass.adapters.inbound.fastapi_.configure import get_configured_app
+from mass.adapters.outbound.aggregator import AggregatorCollection, AggregatorFactory
+from mass.adapters.outbound.dao import DaoCollection
+from mass.config import Config
+from mass.core.query_handler import QueryHandler
+from mass.ports.inbound.query_handler import QueryHandlerPort
+
+
+@asynccontextmanager
+async def prepare_core(*, config: Config) -> AsyncGenerator[QueryHandlerPort, None]:
+    """Constructs and initializes all core components and their outbound dependencies."""
+    dao_factory = MongoDbDaoFactory(config=config)
+    dao_collection = await DaoCollection.construct(dao_factory, config)
+    aggregator_factory = AggregatorFactory(config=config)
+    aggregator_collection = await AggregatorCollection.construct(
+        aggregator_factory, config
```

Comment:
Would be nicer to use kwargs. Would also be great to enforce kwargs directly in the construct method using the "*" before the args.